osbs/tekton.py

```diff
@@ -331,40 +316,15 @@ def log_and_sleep():
                     log_and_sleep()
                     continue
 
-            encoding = None
-
             # Avoid races. We've already asked the server to tell us
             # about changes to the object, but now ask for a fresh
             # copy of the object as well. This is to catch the
             # situation where the object changed before the call to
             # this method, or in between retries in this method.
-            if get_url is not None:
-                logger.debug("retrieving fresh version of object")
-                fresh_response = self.get(get_url)
-                check_response(fresh_response)
-                yield None, fresh_response.json()
-
-            for line in response.iter_lines():
-                logger.debug('%r', line)
-
-                if not encoding:
-                    encoding = guess_json_utf(line)
-
-                try:
-                    j = json.loads(line.decode(encoding))
-                except ValueError:
-                    logger.error("Cannot decode watch event: %s", line)
-                    continue
-
-                if 'object' not in j:
-                    logger.error("Watch event has no 'object': %s", j)
-                    continue
-
-                if 'type' not in j:
-                    logger.error("Watch event has no 'type': %s", j)
-                    continue
-
-                yield (j['type'].lower(), j['object'])
+            logger.debug("retrieving fresh version of object")
+            fresh_response = self.get(get_url)
+            check_response(fresh_response)
+            yield fresh_response.json()
```

Comment:
Wait, how does this method "watch" anything now? It will just just fetch current info about an object WATCH_RETRY times, no?

To be fair, the original method doesn't make a whole lot of sense to me either, but at least it iterates through a streamed response, which I thought was where the "watching" happened

Reply:
Ack, works for me :+1: